marshal.go

```diff
@@ -5,42 +5,66 @@ package fargo
 import (
 	"encoding/json"
 	"encoding/xml"
+	"fmt"
 	"io"
 	"strconv"
 )
 
-// Temporary structs used for GetAppsResponse unmarshalling
-type getAppsResponseArray GetAppsResponse
-type getAppsResponseSingle struct {
-	Application   *Application `json:"application"`
-	AppsHashcode  string       `json:"apps__hashcode"`
-	VersionsDelta int          `json:"versions__delta"`
+func intFromJSONNumberOrString(jv interface{}, description string) (int, error) {
+	switch v := jv.(type) {
+	case float64:
```

Comment:
Per [the documentation for `json.Unmarshal`](https://golang.org/pkg/encoding/json/#Unmarshal), numbers coming from JSON into Go are always of type `float64`—barring use of [`json.(*Decoder).UseNumber`](https://golang.org/pkg/encoding/json/#Decoder.UseNumber). As it stands, _fargo_ does not set this option on a `Decoder` [when decoding a `fargo.GetAppsResponse`](https://github.com/seh/fargo/blob/f1f7c88ad0e3cca61c8a774c5f3c58f3fae53498/marshal.go#L30-L68).